File: android/src/main/kotlin/com/mytiki/sdk/capture/receipt/capacitor/rsp/RspStringType.kt
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

package com.mytiki.sdk.capture.receipt.capacitor.rsp

import com.microblink.core.StringType
import org.json.JSONObject

class RspStringType(stringType: StringType) : Rsp {
    private val confidence: Float
    private val value: String?

    init {
        confidence = stringType.confidence()
        value = stringType.value()
    }

    override fun toJson(): JSONObject =
        JSONObject()
            .put("confidence", confidence)
            .put("value", value)

    companion object {
        fun opt(stringType: StringType?): RspStringType? =
            if (stringType != null) RspStringType(stringType) else null
    }
}

File: ios/Src/Mytiki/Sdk/Capture/Receipt/Capacitor/Rsp/RspStringType.swift
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

import Foundation
import BlinkReceipt
import BlinkEReceipt
import Capacitor

struct RspStringType : Rsp {
    private let confidence: Float
    private let value: String?

    init (stringType: StringType) {
        confidence = stringType.confidence()
        value = stringType.value()
    }
    
    func toJson() -> JSObject {
        JSObject.updateValue("confidence", confidence)
        JSObject.updateValue("value", value)
    }

}
